lib/colorls/core.rb

```diff
@@ -349,18 +349,22 @@ def options(content)
       [key, color, group]
     end
 
-    def tree_traverse(path, prespace, indent)
+    def tree_traverse(path, prespace, depth, indent)
       contents = init_contents(path)
       contents.each do |content|
         icon = content == contents.last || content.directory? ? ' └──' : ' ├──'
         print tree_branch_preprint(prespace, indent, icon).colorize(@colors[:tree])
         print " #{fetch_string(path, content, *options(content))} \n"
         next unless content.directory?
 
-        tree_traverse("#{path}/#{content}", prespace + indent, indent)
+        tree_traverse("#{path}/#{content}", prespace + indent, depth + 1, indent) unless valid_depth?(depth + 1)
```

Comment:
Thinking of a better name -- `valid_depth` does not really match with the thing it does. How about `limit_exceeded?`

And you can just pass `depth` (not `depth + 1`) to the method and use `depth >= @tree[:depth]` for comparison.

Or, for a positive check, applying de Morgan's law:
```
    tree_traverse("#{path}/#{content}", prespace + indent, depth + 1, indent) if keep_going(depth)

def keep_going(depth)
  @tree[:depth].nil? || depth < @tree[:depth]
end
```

Reply:
I vote for the second option 😄 👍